packages/discord.js/src/structures/Integration.js

```diff
@@ -185,6 +185,16 @@ class Integration extends Base {
     } else {
       this.application ??= null;
     }
+
+    if ('scopes' in data) {
+      /**
+       * The scopes this application has been authorized for
+       * @type {?OAuth2Scopes[]}
+       */
+      this.scopes = data.scopes;
```

Comment:
Should we just default this to an array at any time? @discordjs/the-big-4

Reply:
We default many things to an empty array in many places:

- https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/sharding/Shard.js#L43
- https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/ClientApplication.js#L38
- https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/CommandInteractionOptionResolver.js#L42-L50
- https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/Embed.js#L24
- https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/GuildAuditLogsEntry.js#L144
- https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/Message.js#L146
- https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/Message.js#L158
- https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/Presence.js#L235
- https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/SelectMenuInteraction.js#L17

Therefore I believe we should, yes.